src/components/tableview/TableTable.vue

```diff
@@ -392,7 +414,47 @@ export default {
         scrollToStart: false,
         scrollPageUp: false,
         scrollPageDown: false
-      }
+      },
+      tableContextMenu: [
+        {
+          label: '<i class="item-icon material-icons">add_circle_outline</i> Add row',
+          action: () => {
+            this.tabulator.addRow({}, false).then(row => { 
+              this.addRowToPendingInserts(row)
+              this.tabulator.scrollToRow(row, 'bottom', false)
+            })
+          }
+        }
+      ],
+      rowContextMenu:[
+        {
+          label: '<i class="item-icon material-icons">add_circle_outline</i> Add row',
+          action: () => {
+            this.tabulator.addRow({}, false).then(row => { 
+              this.addRowToPendingInserts(row)
+              this.tabulator.scrollToRow(row, 'bottom', false)
+            })
+          }
+        },
+        {
+          label: '<i class="item-icon material-icons">content_copy</i> Clone row',
+          action: (e, row) => {
+            this.tabulator.addRow(row.getData(), false).then(row => {
```

Comment:
Two things here:

- I think one nice thing to do here would be to clear the `primaryKey` column if there is one, as you cannot insert with a duplicate primary key. This should be as easy as `data[this.primaryKey] = undefined`
- Probably `row.getData()` returns the actual object used by the row, so might need to make a copy of it? `const data = { ...row.getData() }`